src/types/AST.ts

```diff
@@ -6,6 +6,8 @@ export type AST = TAny | TArray | TBoolean | TEnum | TInterface | TNamedInterfac
   | TIntersection | TLiteral | TNumber | TNull | TObject | TReference
   | TString | TTuple | TUnion | TCustomType
 
+export type TSuperType = TNamedInterface | TCustomType
```

Comment:
Could there be a way to indicate with the types that `standaloneName` is always defined for supertypes? eg.
```suggestion
export type TSuperType = ASTWithStandaloneName & (TNamedInterface | TCustomType)
```

Reply:
Done